### ISSUE TITLE
Natural Locomotion support

### ISSUE THREAD

**Issue by Eirenliel** [SlimeVR/SlimeVR-Server#25]
To support [Natural Locomotion](https://store.steampowered.com/app/798810/Natural_Locomotion/) SlimeVR need to report speed (or acceleration) values in the driver. This requires support from firmware, software, and driver. Creating issue here for this feature, but it needs to be implemented everywhere.

**Comment by Louka3000** [comment]
What's the status on this?
NaLo no longer crashes... I think all we need to do is forward accel data to SteamVR now, right?
Conveniently, Stermere already did most of the job ;)

**Comment by BlackCube4** [comment]
+1

**Comment by ruti75** [comment]
Hello,
Is there an update on this? I've been trying to write my own code (only using the ankle trackers for simplicoty) and I can share it but I'm a total noob amd was hoping someone did a better job than me.

**Comment by M1DNYT3** [comment]
The changes are merged into the main branch. But **they are not yet in the release**, once they're eventually bundled into the release of the app, the issue should be resolved, and to enable Natural Locomotion support, the toggle in the settings has to be turned on (see the screenshot below). The toggle is off by default, and the setting is persistent, so it'll remember its state between server restarts.

Note that the description mentions the fact that enabling Velocity can introduce jitter in FBT, so it's recommended to only enable Velocity when you're not playing anything where you need smooth visuals of your FBT avatar. The jitter impacts titles like VRChat or BeatSaber's third-person camera with custom FBT avatar mods. The jitter will be looked into, and if there's a minimal, plausible fix, then the note about it will be removed. Otherwise, once the release lands and you get access to this feature, please be aware of this problem.

How it'll look:
<img width="1617" height="892" alt="Screenshot 2026-05-04 153350" src="https://github.com/user-attachments/assets/4da32c96-0fb9-42b5-8fc5-9cbc2ddc310d" />
